### PR TITLE
History object tests

### PR DESCRIPTION
**Pull request by acpaquette**
Adds history object tests for updated history object

## Description
Updates the old history object tests to a gtest and tests the new functionality of the history object

## Related Issue
N/A

## Motivation and Context
Blob refactor and IO improvements

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/Fixtures.h [hunk 1]
```diff
@@ -249,6 +249,14 @@ class CSMCameraDemFixture : public CSMCubeFixture {
 
     void SetUp() override;
 };
+
+class HistoryBlob : public TempTestingFiles {
+  protected:
+    Blob *historyBlob;
```
Comment (jessemapel) [inline isis/tests/Fixtures.h]: This needs to get clean-ed up at the end of each test. I would just change to not be a pointer and assign over it in setup.

File: isis/tests/HistoryTests.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  Blob *blob = history.toBlob();
+
+  std::stringstream os;
+  char *blob_buffer = blob->getBuffer();
+  Pvl newHistoryPvl;
+  for (int i = 0; i < blob->Size(); i++) {
+    os << blob_buffer[i];
+  }
+  os >> newHistoryPvl;
```
Comment (jessemapel) [inline isis/tests/HistoryTests.cpp:44]: I wouldn't interrogate the Blob itself, I would just test History -> Blob -> History works correctly.

**Review by jessemapel — COMMENTED** [review]
Just minor comments. Make sure that you delete the old unit test. It looks to be just commented out anyway.

**Review by AustinSanders — APPROVED** [review]

File: isis/tests/HistoryTests.cpp (hunk 2 above)
Comment (acpaquette) [reply isis/tests/HistoryTests.cpp:44]: Maybe? That's testing a lot in one test, which is why I thought chunking the test up would be better. 

It seems like we either test all of the functionality of the class in one test that is an end to end test, or we chunk it up like I have. In the end to end test you would use the default constructor, the AddEntry method, the toBlob method, the blob constructor and the ReturnHist method to examine the expect PVL

**Review by jessemapel — APPROVED** [review]

**Pull request merged**